src/query.rs

```diff
@@ -206,8 +247,45 @@ impl OverrideStack {
     ///
     /// The inserted overrides will take precedence over any lower item in the stack,
     /// if both maps have a not-`None` entry for a given lint.
-    pub fn push(&mut self, item: &OverrideMap) {
-        self.0.push(item.clone());
+    ///
+    /// # Errors
+    ///
+    /// Returns an `Err` if there is a configuration conflict in the passed map: when
+    /// multiple entries in `item` configure the same entry for the same lint. To handle
+    /// configuration precedence in a defined way, call this function multiple times, and
+    /// entries at the top of the stack will override entries towards the bottom of the stack
+    /// when there is a conflict.
+    pub fn try_push(&mut self, item: &OverrideMap) -> anyhow::Result<()> {
+        let mut compiled_map = BTreeMap::<_, QueryOverride>::new();
+
+        for (id, overrides) in item {
+            let lints = id.lints(ALL_QUERIES.get_or_init(SemverQuery::all_queries));
+
+            for lint in lints {
+                if let Some(entry) = compiled_map.get_mut(lint) {
+                    if let Some(lint_level) = overrides.lint_level {
+                        if let Some(old) = entry.lint_level {
+                            anyhow::bail!("Configuration conflict detected for lint `{lint}`:\n\
```

Comment:
I *think* our bail errors are generally lowercase since `bail!(<stuff>)` gets printed as `error: <stuff>` when the program exits. If I'm right about that, any capitalized error messages should be tested + changed.